### PR TITLE
8204541: Correctly support AIX xlC 16.1 symbol visibility flags

### PR DESCRIPTION
**Pull request by takiguc**
This change is just for AIX platform with IBM XL C/C++ 16.1.
By this change, lib's entry points are reduced by compilation and linkage option changes.

Without changes:
```
$ dump -X32_64 -Tv lib/libawt.so | grep EXP | wc
914
```
With changes
```
$ dump -X32_64 -Tv lib/libawt.so | grep EXP | wc
142 
```

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8204541](https://bugs.openjdk.java.net/browse/JDK-8204541): Correctly support AIX xlC 16.1 symbol visibility flags

### Reviewers
 * [Magnus Ihse Bursie](https://openjdk.java.net/census#ihse) (@magicus - **Reviewer**)
 * [Thomas Stuefe](https://openjdk.java.net/census#stuefe) (@tstuefe - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7862/head:pull/7862` \
`$ git checkout pull/7862`

Update a local copy of the PR: \
`$ git checkout pull/7862` \
`$ git pull https://git.openjdk.java.net/jdk pull/7862/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7862`

View PR using the GUI difftool: \
`$ git pr show -t 7862`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7862.diff">https://git.openjdk.java.net/jdk/pull/7862.diff</a>

</details>

### CODE REVIEW

File: make/autoconf/flags-ldflags.m4 [hunk 1]
```diff
@@ -77,7 +77,8 @@ AC_DEFUN([FLAGS_SETUP_LDFLAGS_HELPER],
         -fPIC"
 
   elif test "x$TOOLCHAIN_TYPE" = xxlc; then
-    BASIC_LDFLAGS="-b64 -brtl -bnorwexec -bnolibpath -bexpall -bernotok -btextpsize:64K \
+    # -bexpall should be used instead of -bnoexpall if -qvisibility option is not defined
```
Comment (magicus) [inline make/autoconf/flags-ldflags.m4:80]: If you hard-code for the assumption that xlc v16 is being used, then this comment does not make any sense..?

Comment (takiguc) [reply make/autoconf/flags-ldflags.m4]: Thanks, @magicus .
I removed the comment.

**Comment by tstuefe** [comment]
So, does this mean we cannot build with older xlC versions anymore? Just curious.

Pinging @backwaterred .

**Comment by takiguc** [comment]
Thanks, @tstuefe .

I saw following link, I 'm using AIX XL C++ 16.1.
https://wiki.openjdk.java.net/display/Build/Supported+Build+Platforms

And following change was already applied. 
[JDK-8224214](https://bugs.openjdk.java.net/browse/JDK-8224214) [AIX] Remove support for legacy xlc compiler

@backwaterred , could you check older compiler should be supported ?

**Comment by tstuefe** [comment]
> Thanks, @tstuefe .
> 
> I saw following link, I 'm using AIX XL C++ 16.1. https://wiki.openjdk.java.net/display/Build/Supported+Build+Platforms
> 
> And following change was already applied. [JDK-8224214](https://bugs.openjdk.java.net/browse/JDK-8224214) [AIX] Remove support for legacy xlc compiler
> 
> @backwaterred , could you check older compiler should be supported ?

Okay, thank you for the clarification. I am fine with this change, I was just curious.

Thanks, Thomas

**Review by magicus — APPROVED** [review]

**Comment by backwaterred** [comment]
Looks good to me :-)

**Review by tstuefe — APPROVED** [review]
Looks good.

**Pull request closed without merging**